influxdb_store.go

```diff
@@ -0,0 +1,409 @@
+	// If no errors query execution was successfully - either DB was created or already exists.
+	response, err := in.con.Query(influxDBClient.Query{
+		Command: fmt.Sprintf("%s %s", "CREATE DATABASE IF NOT EXISTS", dbName),
+	})
+	if err != nil {
+		return err
+	}
+	if response.Error() != nil {
+		return response.Error()
+	}
+	return nil
+}
+
+func (in *InfluxDBStore) executeOneQuery(command string) (*influxDBClient.Result, error) {
+	response, err := in.con.Query(influxDBClient.Query{
+		Command:  command,
+		Database: dbName,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if response.Error() != nil {
+		return nil, response.Error()
+	}
+
+	// Expecting one result, since a single query is executed.
+	if len(response.Results) != 1 {
+		return nil, errors.New("unexpected number of results for an influxdb single query")
+	}
+	return &response.Results[0], nil
+}
+
+func (in *InfluxDBStore) findSpanPoint(ID SpanID) (*influxDBClient.Point, error) {
+	q := fmt.Sprintf(`
+		SELECT * FROM spans WHERE trace_id='%s' AND span_id='%s' AND parent_id='%s' GROUP BY *
+	`, ID.Trace, ID.Span, ID.Parent)
+	result, err := in.executeOneQuery(q)
+	if err != nil {
+		return nil, err
+	}
+	if len(result.Series) == 0 {
+		return nil, nil
+	}
+	if len(result.Series) > 1 {
+		return nil, errors.New("unexpected multiple series")
+	}
+	r := result.Series[0]
+	if len(r.Values) == 0 {
+		return nil, errors.New("unexpected empty series")
+	}
+	p := influxDBClient.Point{
+		Fields: make(pointFields, 0),
+	}
+	fields := r.Values[0]
+	for i, field := range fields {
+		key := r.Columns[i]
+		switch field.(type) {
+		case string:
+			// time field is set by InfluxDB not related to annotations.
+			if key == "time" {
+				t, err := time.Parse(time.RFC3339Nano, field.(string))
+				if err != nil {
+					return nil, err
+				}
+				p.Time = t
+			}
+			if key == "Name" {
+				p.Fields[key] = field.(string)
+			}
+		case nil:
+			continue
+		default:
+			return nil, fmt.Errorf("unexpected time type: %v", reflect.TypeOf(field))
+		}
+	}
+	return &p, err
+}
+
+func (in *InfluxDBStore) init(server *influxDBServer.Server) error {
+	in.server = server
+	url, err := url.Parse(fmt.Sprintf("http://%s:%d", influxDBClient.DefaultHost, influxDBClient.DefaultPort))
+	if err != nil {
+		return err
+	}
+	con, err := influxDBClient.NewClient(influxDBClient.Config{URL: *url})
+	if err != nil {
+		return err
+	}
+	in.con = con
+	if err := in.createDBIfNotExists(); err != nil {
+		return err
+	}
+	in.tracesPerPage = defaultTracesPerPage
+	return nil
+}
+
+func annotationsFromRow(r *influxDBModels.Row) (*Annotations, error) {
+	// Actually an influxDBModels.Row represents a single InfluxDB serie.
+	// r.Values[n] is a slice containing span's annotation values.
+	var fields []interface{}
+	if len(r.Values) == 1 {
+		fields = r.Values[0]
+	}
+
+	// len(r.Values) might be greater than one - meaning there are
+	// some spans to drop, see: InfluxDBStore.Collect(...).
+	// If so last one is picked.
```

Comment:
Thanks for grammar pointing outs :+1: - actually those docs are deprecated now since we have better strategy to rewrite existing spans - see: ddebe983d4c899eb70a65ced97084eb2a4c6b6c1 & e23224c0c2021961a813acf4190c025814bcb79c